ttl: add telemetry

otan opened the pull request:
This commit adds telemetry to row level TTL configuration, as well as
executions of the row level TTL job.

Release note: None

Release justification: low risk high benefit changes to new
functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77108)
<!-- Reviewable:end -->

otan committed:
ttl: add telemetry

This commit adds telemetry to row level TTL configuration, as well as
executions of the row level TTL job.

Release note: None

Release justification: low risk high benefit changes to new
functionality

rafiss reviewed (approved):
> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77108)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rafiss
